edgecast/routedns/group.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Edgecast Inc., Licensed under the terms of the Apache 2.0
+// license. See LICENSE file in project root for terms.
+
+package routedns
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+)
+
+// GetGroup retrieves group information of the provided groupID.
+func (svc *RouteDNSService) GetGroup(
+	params GetGroupParams,
+) (*DnsRouteGroupOK, error) {
+
+	apiURL := fmt.Sprintf(
+		"/v2/mcc/customers/%s/dns/group?id=%d&groupType=%s",
+		params.AccountNumber,
+		params.GroupID,
+		params.GroupProductType.String(),
+	)
+
+	log.Printf("apiURL:%s", apiURL)
+	request, err := svc.Client.BuildRequest("GET", apiURL, nil)
+
+	if err != nil {
+		return nil, fmt.Errorf("GetGroup->Build Request Error: %v", err)
+	}
+
+	parsedResponse := DnsRouteGroupOK{}
+
+	_, err = svc.Client.SendRequest(request, &parsedResponse)
+	if err != nil {
+		return nil, fmt.Errorf("GetGroup->API Response Error: %v", err)
+	}
+	log.Printf("GetGroup->parsedResponse:%v", parsedResponse)
+
+	return &parsedResponse, nil
+}
+
+// AddGroup creates a new load balanced or failover group.
+func (svc *RouteDNSService) AddGroup(params AddGroupParams) (*int, error) {
+	apiURL := fmt.Sprintf(
+		"/v2/mcc/customers/%s/dns/group",
+		params.AccountNumber,
+	)
+
+	request, err := svc.Client.BuildRequest("POST", apiURL, params.Group)
+	if err != nil {
+		return nil, fmt.Errorf("AddGroup->Build Request Error: %v", err)
+	}
+
+	resp, err := svc.Client.SendRequestWithStringResponse(request)
+
+	if err != nil {
+		return nil, fmt.Errorf("AddGroup->API Response Error: %v", err)
+	}
+
+	groupID, err := strconv.Atoi(*resp)
+	if err != nil {
+		return nil, fmt.Errorf(
```

Comment:
Might be helpful to also print out the string text that failed to parse (if the error does not already include it)

Reply:
Tested and confirmed the error does include the string.

e.g. resp, err := strconv.Atoi("notAnum")
result: parsing "notAnum": invalid syntax